fluent/fluent.go

```diff
@@ -386,7 +444,11 @@ func (f *Fluent) run() {
 				emitEventDrainMsg.Do(func() { fmt.Fprintf(os.Stderr, "[%s] Discarding queued events...\n", time.Now().Format(time.RFC3339)) })
 				continue
 			}
-			err := f.write(entry)
+			err := f.write(entry.msg)
+			if entry.result != nil {
```

Comment:
Are there any cases to get `nil` on `entry.result`?

Reply:
Yes, so in this change the sync flow and the async flow get very similar. Both write to the `pending` channel and have a separate go-routine taking entries of that channel and writing the data to the TCP connection. 

In case of the async flow no feedback is required, so there will be no result channel in the entry. As you can see in  `appendBuffer` an entry is written to the channel without a result channel.

In case of the sync flow, a result channel is added to the entry, so the feedback can be returned to the caller. And as you can see `appendBufferBlocking` puts the result channel in the entry and waits on the feedback (which is therefore called "blocking").